src/core/src/sys/compute.rs

```diff
@@ -236,8 +236,8 @@ impl Compute for PressureAtTemperature {
         let virial_tensor = system.virial();
         let virial = virial_tensor.trace();
         let volume = system.volume();
-        let natoms = system.size() as f64;
-        return natoms * K_BOLTZMANN * self.temperature / volume + virial / (3.0 * volume);
+        let nmols = system.molecules().len() as f64;
```

Comment:
OK, I think I got it. The real formula for computing pressure does not include the number of atoms nor the number of molecules, it is the number of degree of freedom over the dimensionality of the system. Which roughly equals the number of atoms for flexible molecules, and the number of molecules for rigid molecules.

[LAMMPS](http://lammps.sandia.gov/doc/compute_pressure.html) documentation explains this quite well, but does not give any algorithm to compute it.

[DL_POLY]() manual page 68 has a "constraint" term in the pressure equation, because in MD rigid molecules only comes from constraint algorithms.

[Tohwee](http://towhee.sourceforge.net/algorithm/pressure.html) uses 4 different algorithms to compute the pressure in MC simulations, I do not know which one would be the best here.

In the [Allen & Tildesley book](https://www.amazon.com/Computer-Simulation-Liquids-Science-Publications/dp/0198556454) page 47, they follow your definition using the number of molecules, whereas in the [Frenkel & Smit](https://www.amazon.com/Understanding-Molecular-Simulation-Second-Computational/dp/0122673514) book page 84, they use the number of atoms.

I do not think that implementing the correct version of the code (using degree of freedom) is doable here, so maybe we can revert this change and indicate clearly in the documentation that this does not work for rigid molecules? We should also open a separated issue to track work on this problem.